hack/generated/pkg/armclient/template_client.go

```diff
@@ -142,6 +143,20 @@ func AuthorizerFromEnvironment() (autorest.Authorizer, error) {
 		return nil, err
 	}
 
+	// the previous never returns an error, so we must do
+	// the checks ourselves…
+	var errs []error
+	requiredEnvVars := []string{auth.SubscriptionID, auth.ClientSecret, auth.ClientID, auth.TenantID}
```

Comment:
do we need to support anything other than client-secret authentication?

Reply:
Eventually we will need to support MSI too.